pyteal/ast/bytes.py

```diff
@@ -29,35 +39,42 @@ def __init__(self, *args: str) -> None:
             ``Bytes("base16", "0x636F6E74656E74")``.
         """
         super().__init__()
-        if len(args) == 1:
-            self.base = "utf8"
-            self.byte_str = escapeStr(args[0])
-        elif len(args) == 2:
-            self.base, byte_str = args
+        if arg2 is None:
+            if type(arg1) is str:
+                self.base = "utf8"
+                self.byte_str = escapeStr(arg1)
+            elif type(arg1) in (bytes, bytearray):
+                self.base = "base16"
+                self.byte_str = cast(Union[bytes, bytearray], arg1).hex()
+            else:
+                raise TealInputError("Unknown argument type: {}".format(type(arg1)))
```

Comment:
Small suggestion: consider using [f-strings](https://www.python.org/dev/peps/pep-0498/) instead of `.format()`

Reply:
Good idea, but this should probably be a separate issue, since `format` is used in so many places in the codebase.

Reply:
Sounds good to me